apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoader.java

```diff
@@ -596,6 +616,240 @@ private void processAsync(
+
+      enrollments.add(
+          getEnrollment(
+              beneficiaryRecord,
+              String.format("%s-%s", date, "10"),
+              beneficiaryRecord.getEntitlementBuyInOctInd(),
+              beneficiaryRecord.getFipsStateCntyOctCode(),
+              beneficiaryRecord.getHmoIndicatorOctInd(),
+              beneficiaryRecord.getMedicaidDualEligibilityOctCode(),
+              beneficiaryRecord.getMedicareStatusOctCode(),
+              beneficiaryRecord.getPartCContractNumberOctId(),
+              beneficiaryRecord.getPartCPbpNumberOctId(),
+              beneficiaryRecord.getPartCPlanTypeOctCode(),
+              beneficiaryRecord.getPartDContractNumberOctId(),
+              beneficiaryRecord.getPartDLowIncomeCostShareGroupOctCode(),
+              beneficiaryRecord.getPartDPbpNumberOctId(),
+              beneficiaryRecord.getPartDRetireeDrugSubsidyOctInd(),
+              beneficiaryRecord.getPartDSegmentNumberOctId()));
+
+      enrollments.add(
+          getEnrollment(
+              beneficiaryRecord,
+              String.format("%s-%s", date, "11"),
+              beneficiaryRecord.getEntitlementBuyInNovInd(),
+              beneficiaryRecord.getFipsStateCntyNovCode(),
+              beneficiaryRecord.getHmoIndicatorNovInd(),
+              beneficiaryRecord.getMedicaidDualEligibilityNovCode(),
+              beneficiaryRecord.getMedicareStatusNovCode(),
+              beneficiaryRecord.getPartCContractNumberNovId(),
+              beneficiaryRecord.getPartCPbpNumberNovId(),
+              beneficiaryRecord.getPartCPlanTypeNovCode(),
+              beneficiaryRecord.getPartDContractNumberNovId(),
+              beneficiaryRecord.getPartDLowIncomeCostShareGroupNovCode(),
+              beneficiaryRecord.getPartDPbpNumberNovId(),
+              beneficiaryRecord.getPartDRetireeDrugSubsidyNovInd(),
+              beneficiaryRecord.getPartDSegmentNumberNovId()));
+
+      enrollments.add(
+          getEnrollment(
+              beneficiaryRecord,
+              String.format("%s-%s", date, "12"),
+              beneficiaryRecord.getEntitlementBuyInDecInd(),
+              beneficiaryRecord.getFipsStateCntyDecCode(),
+              beneficiaryRecord.getHmoIndicatorDecInd(),
+              beneficiaryRecord.getMedicaidDualEligibilityDecCode(),
+              beneficiaryRecord.getMedicareStatusDecCode(),
+              beneficiaryRecord.getPartCContractNumberDecId(),
+              beneficiaryRecord.getPartCPbpNumberDecId(),
+              beneficiaryRecord.getPartCPlanTypeDecCode(),
+              beneficiaryRecord.getPartDContractNumberDecId(),
+              beneficiaryRecord.getPartDLowIncomeCostShareGroupDecCode(),
+              beneficiaryRecord.getPartDPbpNumberDecId(),
+              beneficiaryRecord.getPartDRetireeDrugSubsidyDecInd(),
+              beneficiaryRecord.getPartDSegmentNumberDecId()));
+
+      beneficiaryRecord.setEnrollments(enrollments);
```

Comment:
I think this will overwrite **all** enrollment records, instead of just the current year's won't it? If you want to avoid that, you need to merge the new/current records into the existing list, avoiding duplicates (which would likely cause PK errors).

More importantly, if this wasn't caught by your tests, then I think you need more ITs to ensure that all of this is actually accomplishing the thing we need it to.

Reply:
Yea Changed the logic and adding more IT's